pkg/logql/log/fmt.go

```diff
@@ -150,12 +173,54 @@ func (lf *LabelsFormatter) Process(l []byte, lbs *LabelsBuilder) ([]byte, bool)
 			continue
 		}
 		lf.buf.Reset()
-		//todo (cyriltovena): handle error
 		if data == nil {
 			data = lbs.Labels().Map()
 		}
-		_ = f.tmpl.Execute(lf.buf, data)
+		if err := f.tmpl.Execute(lf.buf, data); err != nil {
+			lbs.SetErr(errTemplateFormat)
+			continue
+		}
 		lbs.Set(f.Name, lf.buf.String())
 	}
 	return l, true
 }
+
+func trunc(c int, s string) string {
+	if c < 0 && len(s)+c > 0 {
```

Comment:
we support negative truncation, I'm basically mimicking the sprig library which is the same as helm template.

I'm going to support runes as you said, since the original library don't, but I won't change the behaviour.

I'm adding test too.